api/buffer/design.go

```diff
@@ -0,0 +1,542 @@
+	})
+	Attribute("content", String, func() {
+		MinLength(1)
+		MaxLength(4096)
+	})
+	Attribute("dataType", String, func() {
+		Enum("STRING", "INTEGER", "NUMERIC", "FLOAT", "BOOLEAN", "DATE", "TIMESTAMP")
+		Default("STRING")
+	})
+	Required("language", "undefinedValueStrategy", "content")
+})
+
+var ImportConditions = Type("Conditions", func() {
+	Description("Table import triggers.")
+	Attribute("count", Int, func() {
+		Description("Maximum import buffer size in number of records.")
+		Minimum(1)
+		Maximum(10_000_000)
+		Default(1000)
+	})
+	Attribute("size", Int, func() {
+		Description("Maximum import buffer size in bytes.")
+		Minimum(100)
+		Maximum(50_000_000)
+		Default(5_000_000)
+	})
+	Attribute("time", Int, func() {
+		Description("Minimum import interval in seconds.")
+		Minimum(30)
+		Maximum(24 * 60 * 60)
+		Default(5 * 60)
+	})
+})
+
+// Errors ------------------------------------------------------------------------------------------------------------
+
+var GenericErrorType = Type("GenericError", func() {
+	Description("Generic error")
+	Attribute("statusCode", Int, "HTTP status code.", func() {
+		Example(StatusInternalServerError)
+	})
+	ErrorName("error", String, "Name of error.", func() {
+		Meta("struct:field:name", "name")
+		Example("templates.internalError")
+	})
+	Attribute("message", String, "Error message.", func() {
+		Example("Internal Error")
+	})
+	Required("statusCode", "error", "message")
+})
+
+func GenericError(statusCode int, name, description, example string) {
+	// Must be called inside HTTP definition
+	endpoint, ok := eval.Current().(*expr.HTTPEndpointExpr)
+	if !ok {
+		eval.IncompatibleDSL()
+	}
+
+	// Add error to the method definition
+	eval.Execute(func() {
+		Error(name, GenericErrorType, func() {
+			Description(description)
+			Example(ExampleError(statusCode, name, example))
+		})
+	}, endpoint.MethodExpr)
+
+	// Add response to the HTTP method definition
+	Response(name, statusCode)
+}
+
+func ReceiverNotFoundError() {
+	GenericError(StatusNotFound, "buffer.receiverNotFound", "Receiver not found error.", `Receiver "github-pull-requests" not found.`)
+}
+
+func ExportNotFoundError() {
+	GenericError(StatusNotFound, "buffer.exportNotFound", "Export not found error.", `Export "github-changed-files" not found.`)
+}
+
+// Examples ------------------------------------------------------------------------------------------------------------
+
+func ExampleError(statusCode int, name, message string) map[string]interface{} {
+	return map[string]interface{}{
+		"statusCode": statusCode,
+		"error":      name,
+		"message":    message,
+	}
+}
+
+func exampleReceiver() map[string]interface{} {
+	id := "github-pull-requests"
+	return map[string]interface{}{
+		"id":      &id,
+		"url":     "https://buffer.keboola.com/v1/import/github-pull-requests/#/UBdJHwifkaQxbVwPyaRstdYpcboGwksSluCGIUWKttTiUdVH",
+		"exports": exampleExportArray(),
+	}
+}
+
+func exampleExportArray() []map[string]interface{} {
```

Comment:
Použil jsem tady `map[string]interface{}`, abych nemusel ztrávit zbytek života u errorů typu "example value ... is incompatible attribute", který mi ale neřeknou, proč jsou ty typy incompatible 🙂.